.github/workflows/unit-tests.yaml

```
@@ -0,0 +1,28 @@
+name: encrypt-cloud-image unit tests
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.18'
```

```suggestion
      - name: Set up Go
        uses: actions/setup-go@v5
        with:
          go-version-file: './go.mod'
```

This will download the version of Go specified in the go.mod

---

I saw multiple failures with go 1.13 which I couldn't interpret since they were about functions not existing because they were introduced in later versions. I am not sure why that happened though. I implemented the suggested change to check again.

---

Check https://github.com/canonical/encrypt-cloud-image/actions/runs/13199519664/job/36848169345?pr=29